AxonDeepSeg/segment.py

```diff
@@ -108,10 +139,38 @@ def segment_image(
 
         img_name_original = acquisition_name.stem
 
+        # Get the model's native resolution
+        model_resolution, patch_size = get_model_native_resolution_and_patch(path_model)
+
+        # Check that the resampled image will be of sufficient size, and if not throw an error.
+        im = Image.open(path_testing_image)
+        w, h = im.size
+
+        w_resampled = w*(acquired_resolution*zoom_factor)/model_resolution
+        h_resampled = h*(acquired_resolution*zoom_factor)/model_resolution
```

Comment:
The "/" is problematic here in case there is a mix between float values (isotropic acquired_resolution) and list values (anisotropic model_resolution)
```
TypeError: unsupported operand type(s) for /: 'float' and 'list'
```

Reply:
I was thinking about that, but felt that programming for the anisotropic case here now was going overboard, as I still have yet to have a model/test image to work with during development. It is difficult to continue supporting an edge case like this only in abstract terms. If you can provide me with a model trained on anisotropic image and a test image to go with it, then I think we could continue supporting this edge case, as then I can write tests for it, otherwise, I'd simply write an error/exception message that we don't support it yet through an if/else.

Reply:
Maybe we could fix this by adding an else clause in `get_model_native_resolution_and_patch()` like so:
```python
if model_resolution[0] == model_resolution[1]: #isotropic pixels
   model_resolution = model_resolution[0]
else:
   model_resolution = max(model_resolution)
```
As for the anisotropic `acquired_resolution` we would need to check it before applying this formula.

Reply:
I'm not sure that would work because it's the combination of "model resolution" and "patch_size" for a _given dimension_ that is problematic, not only the max model_resolution and min patch_size of the 2 dimensions.

My point was more that the current implementation is not consistent between the model_resolution and patch_size assignement on lines 97-100 (accepting both isotropic and anisotropic) with the computation here that assumes isotropic.

I agree with Mathieu that we are not dealing with anisotropic pixels elsewhere in ADS, so throwing an error here for the moment would be sufficient.

Reply:
The `def get_model_native_resolution_and_patch` function now throws an error if the model uses an anisotropic pixel size, as a temporary fix until we decide to add support for it. I tried it with an anisotropic resolution in the model json config and it works for now.